internal/pkg/template/diff/diff.go

```diff
@@ -19,10 +21,107 @@ type Node struct {
 
 // Parse constructs a diff tree that represent the differences between two YAML documents.
 func Parse(curr, old []byte) (*Node, error) {
```

Comment:
now that I think about it, what do you think of this API format instead: https://dave.cheney.net/2019/09/24/be-wary-of-functions-which-take-several-parameters-of-the-same-type

```go
type From []byte

// Diff constructs a tree of *Node that represents the differences between the from and to YAML documents.
func (from From) Diff(to []byte) (*Node, error)
```

Reply:
done!